### PR TITLE
Cleanup / upgrade packages

### PR DESCRIPTION
**Pull request by umidbekkarimov**
Remove unused / upgrade old dependencies.

### CODE REVIEW

**Comment by umidbekkarimov** [comment]
Also add `ts-jest` and `npm-run-all`.
After this updates I will start working on with #1111

**Comment by leebyron** [comment]
There's a bit too much happening in this PR. do you mind separating out the stuff unrelated to package.json dependencies and yarn.lock?

**Comment by leebyron** [comment]
Specifically the ts-jest changes are controversial and reduce test coverage, so I'd like to discuss those in isolation

**Comment by umidbekkarimov** [comment]
> There's a bit too much happening in this PR. do you mind separating out the stuff unrelated to package.json dependencies and yarn.lock?

Yep. Sorry, I just can't stop sometimes.

**Comment by umidbekkarimov** [comment]
> Specifically the ts-jest changes are controversial and reduce test coverage, so I'd like to discuss those in isolation

Oh, it was recommended in jest [readme](https://github.com/facebook/jest#using-typescript) so i thought it's ok.

File: package.json [hunk 1]
```diff
@@ -77,25 +72,24 @@
     "gulp-sourcemaps": "1.6.0",
     "gulp-uglify": "2.0.1",
     "gulp-util": "3.0.8",
-    "jasmine-check": "^0.1.2",
-    "jest": "^17.0.3",
-    "magic-string": "0.10.2",
+    "jasmine-check": "0.1.5",
+    "jest": "19.0.2",
     "marked": "0.3.6",
-    "microtime": "^2.0.0",
```
Comment (leebyron) [inline package.json]: This one is actually an optionalDep of benchmark, so I'm adding this back in a follow-up commit

**Pull request merged**